modin/pandas/test/utils.py

```diff
@@ -730,13 +730,23 @@ def applyier(module, *args, **kwargs):
         result = getattr(module, fn_name)(*args, **kwargs)
         if cast_to_str:
             result = result.astype(str)
+            if isinstance(result.index, pandas.MultiIndex):
+                levels_number = len(result.index.levels)
+                new_index_data = [
+                    result.index.get_level_values(lev_id).astype(str)
+                    for lev_id in range(levels_number)
+                ]
+                new_index = pandas.MultiIndex.from_arrays(new_index_data)
+                new_index.names = result.index.names
+                result.index = new_index
```

Comment:
why not just `result.index = result.index.astype(str)`?

Reply:
If we apply `astytpe` to the `Multiindex` object directly we will obtain error
`TypeError: Setting <class 'pandas.core.indexes.multi.MultiIndex'> dtype to anything other than object is not supported`
But actually in this PR `Multiindex` conversion is not used, so these changes were removed as unrelated.

Reply:
So are you going to delete this?